test/e2e/l7networkpolicy_test.go

```diff
@@ -289,6 +293,77 @@ func testL7NetworkPolicyHTTP(t *testing.T, data *TestData) {
 	})
 }
 
+func testL7NetworkPolicyHTTPWithLargeBody(t *testing.T, data *TestData) {
+	clientPodName := "test-l7-http-client-selected"
+	clientPodLabels := map[string]string{"test-l7-http-e2e": "client"}
+
+	// Create a client Pod on a Node, with the Pod being selected by the test L7 NetworkPolicy as target.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithLabels(clientPodLabels).Create(data))
+	_, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", clientPodName)
+
+	// Create a hostNetwork server Pod on another Node, with the Pod being selected by the test NetworkPolicy as the
+	// destination.
+	serverPodName := "test-l7-http-server"
+	serverPodLabels := map[string]string{"test-l7-http-e2e": "server"}
+	cmd := []string{"/agnhost", "netexec", "--http-port=8081"}
+	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).
+		WithHostNetwork(true).
+		OnNode(nodeName(1)).
+		WithCommand(cmd).
+		WithLabels(serverPodLabels).
+		Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", serverPodName)
+	serverIPs := podIPs.AsSlice()
+
+	l7ProtocolAllowsPathShell := []crdv1beta1.L7Protocol{
+		{
+			HTTP: &crdv1beta1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/shell*",
+			},
+		},
+	}
+	policyAllowPathShellName := "test-l7-http-allow-path-shell"
+	createL7NetworkPolicy(t, data, true, policyAllowPathShellName, 1, clientPodLabels, serverPodLabels, ProtocolTCP, p8081, l7ProtocolAllowsPathShell)
+	time.Sleep(networkPolicyDelay)
+
+	for _, ip := range serverIPs {
+		baseURL := net.JoinHostPort(ip.String(), strconv.Itoa(int(p8081)))
+		// Verify that the test L7 NetworkPolicy allows access to the "/shell" path with large body payload.
+		assert.Eventually(t, func() bool {
+			// Get the MTU of the test client Pod, assuming it's the MTU of the K8s cluster.
+			cmd := []string{"cat", "/sys/class/net/eth0/mtu"}
+			mtuStdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
+			if err != nil {
+				return false
+			}
+			mtu, err := strconv.Atoi(strings.TrimSpace(mtuStdout))
+			if err != nil {
+				return false
+			}
+
+			// Run the command that makes the test server send an HTTP response with a body larger than the MTU on the
+			// test client Pod.
+			testBodySize := mtu * 2
+			cmd = []string{"sh", "-c", fmt.Sprintf(`curl --data-urlencode "cmd=head -c %d </dev/zero | tr '\0' 'A'" http://%s/shell?cmd`, testBodySize, baseURL)}
```

Comment:
I don't think you need to wrap this command using a shell invocation. Does the following work?
```go
cmd := []string{"curl", "--data-urlencode", fmt.Sprintf("cmd=head -c %d </dev/zero | tr '\0' 'A'", testBodySize), "http://%s/shell?cmd"}
```

Note that while the `cmd` parameter is a shell command, it will be executed on the server, not on the client Pod.

Reply:
I have verified this cmd manually and it can work.